### PR TITLE
feat: move TaskQueue from deno_core and make unsync

### PR DESCRIPTION
**Pull request by dsherret**
Rewrite of https://github.com/dsherret/deno_core/blob/main/core/task_queue.rs

### CODE REVIEW

File: src/task_queue.rs [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+use std::task::Waker;
+
+use crate::Flag;
+
+#[derive(Debug, Default)]
+struct TaskQueueTaskWaker {
+  is_ready: Flag,
+  is_future_dropped: Flag,
+  waker: RefCell<Option<Waker>>,
+}
```
Comment (dsherret) [inline src/task_queue.rs:15]: I'm not really that familiar with wakers and poll in rust, so please review carefully.

File: src/task_queue.rs [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+    assert!(flag.is_raised());
+
+    // now, drop the futures
+    drop(futures);
+
+    assert!(delayed_task.await.unwrap());
+  }
+
+  #[tokio::test]
+  async fn acquires_position_synchronously() {
```
Comment (dsherret) [inline src/task_queue.rs:316]: Unlike the previous implementation, this implementation allows acquiring the position synchronously.

File: Cargo.toml [hunk 3]
```diff
@@ -9,7 +9,7 @@ description = "A collection of adapters to make working with Tokio single-thread
 readme = "README.md"
 
 [dependencies]
-tokio = { version = "1", features = ["rt"] }
+tokio = { version = "1", features = ["macros", "rt"] }
```
Comment (dsherret) [inline Cargo.toml:15]: I'm not sure how to enable this just for the tests.

**Review by mmastrac — APPROVED** [review]
LGTM

**Pull request merged**